### PR TITLE
Add classes

### PR DESCRIPTION
**Pull request by Fikerte-T**
- Implement a parent class `person`.
- Implement the classes to represent `students` and `teachers` which are sub-classes that inherit from `person`.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    return true if of_age? || parent_permission
+  end
```
Comment (andersonlebon) [inline person.rb]: - [ ] Kindly note that At this level there is the variable `parent_permission` is not defined. to access it kindly use the instance variable `@parent_permission`
- [ ]  Kindly note also that you don't have an else condition. This method will not return false if the condition did not match. Kindly and an else statement or use this syntax below :point_down: 

```.ruby
 def can_use_services?
    of_age? || @parent_permission
  end
```
or
```rb
 def can_use_services?
    return of_age? || @parent_permission
end
```

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  private
+
+  def of_age?
+    return true if is_of_age? >= 18
+  end
+end
```
Comment (andersonlebon) [inline person.rb]: - [ ] Kindly note that you don't have any method called `is_of_age?` in your class. This function should return true if the age stored in the instance variable `@age` is >= 18. example: :point_down: 

```rb 
 def iof_age?
    return true if @age >= 18
  end
```

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    return true if of_age? || parent_permission
```
Comment (andersonlebon) [inline person.rb]: ## [OPTIONAL]

- [ ] Note that ruby always returns the last line of code in a method. I would suggest removing the `return` keyword as ruby will always do it for you.

File: person.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  private
+
+  def of_age?
+    return true if is_of_age? >= 18
```
Comment (andersonlebon) [inline person.rb]: ## [OPTIONAL]

- [ ] Note that ruby always returns the last line of code in a method. I would suggest removing the `return` keyword as ruby will always do it for you.

File: student.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Student < Person
+  def initialize(classroom)
+    super()
+    @classroom = classroom
+  end
```
Comment (andersonlebon) [inline student.rb]: - [ ] Kindly note that as this class enherits from Persone class its constructor should extend all parent's @instance variables comment in the children initialize parameter. :point_down: 

```rb
class Student < Person
   def initialize(age, classroom, name = "Unknown", parent_permission = true)  
        super(age, name, parent_permission)
          @classroom = classroom
    end
end
```

File: teacher.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Teacher < Person
+  def initialize(specialization)
+    super()
+    @specialization = specialization
+  end
```
Comment (andersonlebon) [inline teacher.rb]: #### The same here :point_down: 

- [ ] Kindly note that as this class enherits from Persone class its constructor should extend all parent's @instance variables comment in the children initialize parameter. :point_down: 

```rb
class Teacher < Person
    def initialize(age, specialization, name = "Unknown", parent_permission = true)
       super(age, name, parent_permission)
       @specialization = specialization
     end
end
```

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Fikerte-T 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Student < Person
+  def initialize(classroom)
```
Comment (Ghiftee) [inline student.rb]: - [ ] Like the previous reviewer mentioned, note that as this class inherits from Person class its constructor should extend all parent's instance variables so kindly add the rest of the variables to initialize
``` def initialize(age, classroom, name = "Unknown", parent_permission = true)  ```

File: teacher.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Teacher < Person
+  def initialize(specialization)
```
Comment (Ghiftee) [inline teacher.rb]: - [ ] Like the previous reviewer mentioned, note that as this class inherits from Person class its constructor should extend all parent's instance variables so kindly add the rest of the variables to initialize
``` def initialize(age, specialization, name = "Unknown", parent_permission = true)  ```

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hi there 👋🏽 , 

Great work on making the changes requested by a previous reviewer 👏 
To highlight 👍🏽 
- The else statement has been added
- is0fage function has been added
- Return keyword has been removed
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @Fikerte-T, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

**Pull request merged**